iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -48,12 +50,18 @@ LogicalResult verifyTensorToVectorsPassPipelineConfig(
   // Verify that the workload per workgroup is set and is non-zero.
   SmallVector<int64_t> workloadPerWorkgroup =
       translationInfo.getWorkloadPerWorkgroupVals();
-  SmallVector<unsigned> partitionedLoops = getPartitionedLoops(op);
-  if (workloadPerWorkgroup.size() != partitionedLoops.size()) {
-    return op->emitOpError("expected ")
-           << partitionedLoops.size()
-           << " entries for workload_per_wg, but got "
-           << workloadPerWorkgroup.size();
+  if (workloadPerWorkgroup.size() > kNumMaxParallelDims) {
+    return op->emitOpError("workload_per_wg size should be less than ")
+           << kNumMaxParallelDims;
+  }
+  if (isa<linalg::LinalgOp, IREE::LinalgExt::TiledOpInterface>(op)) {
+    SmallVector<unsigned> partitionedLoops = getPartitionedLoops(op);
```

Comment:
This utility function also has a case for MMT4D. Maybe rather than doing all the same checks here, it would be better to have the utility function indicate when it can't determine the partitioned loops (rather than returning an empty vector). `FailureOr` or `Optional`?

Reply:
MMT4D is a `LinalgOp`. So its covered. Good point about `FailureOr`. Thats usable, but this is probably going to change to not sue this method and use an interface for it (either me or Lei are going to add this). So maybe leave it as is for now, and do a follow up to get rid of this method altogether?

Reply:
SGTM